src/v/cloud_storage/access_time_tracker.cc

```diff
@@ -155,9 +163,14 @@ ss::future<> access_time_tracker::read(ss::input_stream<char>& in) {
 
 void access_time_tracker::add_timestamp(
   std::string_view key, std::chrono::system_clock::time_point ts) {
+    if (!should_track(key)) {
```

Comment:
Should we also check in the remove_timestamp path, to avoid doing a O(lnN) search for the things we know we never added to begin with?  or does the cache just never call remove_timestamp for non-segment files?

Reply:
I allowed removal of `tx` and `index` keys on purpose. Since we only serialise the hash, we cannot drop all of them when de-serialising. When the tracker is hydrated, it's going to contain such keys which are only ever removed by calls to `access_time_tracker::trim`. The idea is that on the start-up clean we do not apply the filtering and trim away all of these entries we don't need. For subsequent clean-ups, the filtering is applied.